File: frontend/src/services/categoryService.js
import axios from "axios";

const API_URL = "http://localhost:8088/api/categories";

const categoryService = {
  getAllCategories: () => axios.get(`${API_URL}/all`),
  getCategoryById: (id) => axios.get(`${API_URL}/get/${id}`),
  getCategoriesByType: (type) => axios.get(`${API_URL}/type/${type}`),
  createCategory: (category) => axios.post(`${API_URL}/create`, category),
  updateCategory: (id, category) => axios.put(`${API_URL}/update/${id}`, category),
  deleteCategory: (id) => axios.delete(`${API_URL}/delete/${id}`)
};

export default categoryService;

File: Backend/src/main/java/com/service/CategoryService.java
package com.service;

import java.util.List;

import com.model.Category;

public interface CategoryService {
	
	
  public  Category createCategory(Category category);

    
  public  List<Category> getAllCategories();

    
  public  Category getCategoryById(Long id);

    
  public  Category updateCategory(Long id, Category category);

    
  public  void deleteCategory(Long id);

    
   public List<Category> getCategoriesByType(String type);

    
}
